build/Targets/RepoToolset/Sign.proj

```diff
@@ -0,0 +1,75 @@
+<!-- All Rights Reserved. Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE file in the project root for more information. -->
+<Project DefaultTargets="Sign">
+  <Import Project="BuildStep.props" />
+
+  <Import Project="$(NuGetPackageRoot)microsoft.dotnet.signtool\$(MicrosoftDotNetSignToolVersion)\build\Microsoft.DotNet.SignTool.props" />
+
+  <Target Name="Sign">
+    <Error Text="The value of DotNetSignType is invalid: '$(DotNetSignType)'"
+           Condition="'$(DotNetSignType)' != 'real' and '$(DotNetSignType)' != 'test' and '$(DotNetSignType)' != ''" />
+
+    <PropertyGroup>
+      <DryRun>true</DryRun>
+      <DryRun Condition="'$(OfficialBuild)' == 'true'">false</DryRun>
+
+      <TestSign>false</TestSign>
+      <TestSign Condition="'$(DotNetSignType)' == 'test'">true</TestSign>
+
+      <DesktopMSBuildRequired>false</DesktopMSBuildRequired>
+      <DesktopMSBuildRequired Condition="'$(DryRun)' != 'true' and '$(MSBuildRuntimeType)' == 'Core'">true</DesktopMSBuildRequired>
```

Comment:
Should all of these be `_` prefixed since they are local to the file?

Reply:
This is a project file, not a targets file so it won't be imported anywhere where it could cause conflicts.

Reply:
Oh yes good point

---
In reply to: [215462898](https://github.com/dotnet/roslyn/pull/29622#discussion_r215462898) [](ancestors = 215462898)